### PR TITLE
feat: add artifact streaming CSE changes for 22.04

### PR DESCRIPTION
**Pull request by ganeshkumarashok**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/cse_config.sh [hunk 1]
```diff
@@ -340,6 +345,15 @@ ensureTeleportd() {
     systemctlEnableAndStart teleportd || exit $ERR_SYSTEMCTL_START_FAIL
 }
 
+ensureArtifactStreaming(){
+    touch /etc/default/artifact-streaming.test
+    echo "in ensureArtifactStreaming"
+    if [[ "${UBUNTU_RELEASE}" == "22.04" && ( "${CPU_ARCH}" == "amd64" ) ]]; then
```
Comment (cameronmeissner) [inline parts/linux/cloud-init/artifacts/cse_config.sh:351]:
```suggestion
    if [[ "${UBUNTU_RELEASE}" == "22.04" ]]  && [[ "${CPU_ARCH}" == "amd64" ]]; then
```

File: e2e/template.go [hunk 2]
```diff
@@ -351,6 +351,7 @@ func baseTemplate() *datamodel.NodeBootstrappingConfiguration {
 		EnableNvidia:                   false,
 		EnableACRTeleportPlugin:        false,
 		TeleportdPluginURL:             "",
+		EnableArtifactStreaming:        true,
```
Comment (cameronmeissner) [inline e2e/template.go:354]: is this something we want enabled by default for all e2e tests? I'd maybe lean more towards defaulting to false and having specific scenarios for this to be set to true within (though we do have a fair number of scenarios now... I've actually been thinking about adding the concept of "suites" which would be pre-configured scenario-sets designed to test certain things)

Comment (cameronmeissner) [reply e2e/template.go:354]: compromise could also be that we start with false for now and later once this has rolled out we can default it to true

File: parts/linux/cloud-init/artifacts/cse_config.sh [hunk 3]
```diff
@@ -306,6 +306,11 @@ ensureContainerd() {
   if [ "${TELEPORT_ENABLED}" == "true" ]; then
     ensureTeleportd
   fi
+  if [ "${ARTIFACT_STREAMING_ENABLED}" == "true" ]; then
+    echo "ARTIFACT STREAMING is enabled, running test function for artifact streaming"
+    ensureArtifactStreaming
```
Comment (cameronmeissner) [inline parts/linux/cloud-init/artifacts/cse_config.sh:311]:
```suggestion
    logs_to_events ensureArtifactStreaming
```

maybe we want to log this as an actual CSE step?

File: parts/linux/cloud-init/artifacts/cse_config.sh [hunk 4]
```diff
@@ -340,6 +345,15 @@ ensureTeleportd() {
     systemctlEnableAndStart teleportd || exit $ERR_SYSTEMCTL_START_FAIL
 }
 
+ensureArtifactStreaming(){
+    touch /etc/default/artifact-streaming.test
+    echo "in ensureArtifactStreaming"
+    if [[ "${UBUNTU_RELEASE}" == "22.04" && ( "${CPU_ARCH}" == "amd64" ) ]]; then
+        sudo /opt/acr/tools/overlaybd/install.sh || exit $ERR_ARTIFACT_STREAMING_DOWNLOAD_INSTALL
```
Comment (cameronmeissner) [inline parts/linux/cloud-init/artifacts/cse_config.sh:352]: could maybe even call `logs_to_events` on these lines with "AKS.CSE.ensureArtifactStreaming.overlaybdInstall" or something, some with enable-http-auth.sh

File: parts/linux/cloud-init/artifacts/cse_config.sh [hunk 5]
```diff
@@ -340,6 +345,15 @@ ensureTeleportd() {
     systemctlEnableAndStart teleportd || exit $ERR_SYSTEMCTL_START_FAIL
 }
 
+ensureArtifactStreaming(){
```
Comment (cameronmeissner) [inline parts/linux/cloud-init/artifacts/cse_config.sh:348]:
```suggestion
ensureArtifactStreaming() {
```

**Comment by ganeshkumar5699** [comment]
This PR is currently in the testing phase - not ready for review

File: e2e/template.go (hunk 2 above)
Comment (ganeshkumar5699) [reply e2e/template.go:354]: Agreed. This is for ab e2e testing now -- PR is not yet ready for review.

**Pull request closed without merging**